pkg/tcpip/network/fragmentation/reassembler.go

```diff
@@ -116,3 +131,23 @@ func (r *reassembler) checkDoneOrMark() bool {
 	r.mu.Unlock()
 	return prev
 }
+
+// TimeOut function generates ICMP TTL Error message (Fragment reassembly time exceeded message).
+func (r *reassembler) TimeOut(ra *stack.Route, netHeader buffer.View, vv buffer.VectorisedView) {
+	vv = vv.Clone(nil)
+	hdr := buffer.NewPrependable(int(ra.MaxHeaderLength()) + header.ICMPv4MinimumSize + header.IPv4MinimumSize)
+
+	ip_hdr := hdr.Prepend(header.IPv4MinimumSize)
+	copy(ip_hdr, netHeader)
+
```

Comment:
why are we copying the inbound netHEader into the outbound packet. Wont' the source and address be swapped around?

Reply:
ICMP error message supposed to be sent should carry the same header present in ipv4 fragment.